### PR TITLE
Port Keypoint detection to backend agnostic-keras-core

### PR DESCRIPTION
**Pull request by anas-rz**
**NOTE**: This example needs to fill out a Google Form for data
Update imports
Needed to initialize super class in keras.utils.Sequence subclass.
[Colab Notebook](https://colab.research.google.com/drive/1mfLUJfo5tIuYuQtQPN0h23EY3aSi8YK8?usp=sharing)
[keypoint_detection_keras.diff.txt](https://github.com/keras-team/keras-core/files/12098661/keypoint_detection_keras.diff.txt)

### CODE REVIEW

File: examples/keras_io/vision/keypoint_detection.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+visualize_keypoints(images, keypoints)
+
+"""
+The plots show that we have images of non-uniform sizes, which is expected in most
+real-world scenarios. However, if we resize these images to have a uniform shape (for
+instance (224 x 224)) their ground-truth annotations will also be affected. The same
+applies if we apply any geometric transformation (horizontal flip, for e.g.) to an image.
+Fortunately, `imgaug` provides utilities that can handle this issue.
+In the next section, we will write a data generator inheriting the
+[`keras.utils.Sequence`](https://keras.io/api/utils/python_utils/#sequence-class) class
```
Comment (fchollet) [inline examples/keras_io/vision/keypoint_detection.py:244]: Convert `Sequence` to `PyDataset`. For performance you probably want to set `workers`.

File: examples/keras_io/vision/keypoint_detection.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+"""
+To know more about how to operate with keypoints in `imgaug` check out
+[this document](https://imgaug.readthedocs.io/en/latest/source/examples_keypoints.html).
+"""
+
+"""
+## Define augmentation transforms
+"""
+
+train_aug = iaa.Sequential(
```
Comment (fchollet) [inline examples/keras_io/vision/keypoint_detection.py:325]: Something equivalent can be implemented using Keras preprocessing layers -- I suggest we do that. We can use Resizing, RandomFlip, RandomZoom, RandomRotation.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/keras_io/vision/keypoint_detection.py (hunk 2 above)
Comment (anas-rz) [reply examples/keras_io/vision/keypoint_detection.py:339]: The Keras layers don't handle for augmenting key points. We can only use imgaug for this.

Comment (fchollet) [reply examples/keras_io/vision/keypoint_detection.py:339]: Ok

File: examples/keras_io/vision/keypoint_detection.py [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+"""
+
+"""
+## Prepare data generator
+"""
+
+
+class KeyPointsDataset(keras.utils.PyDataset):
+    def __init__(self, image_keys, aug, batch_size=BATCH_SIZE, train=True):
+        super().__init__()
```
Comment (fchollet) [inline examples/keras_io/vision/keypoint_detection.py:261]: Should we pass `workers` here? Otherwise the preprocessing will not be parallelized.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**